web_widget_many2one_polymorphic/static/src/js/web_widget_many2one_polymorphic.js

```diff
@@ -0,0 +1,44 @@
+//-*- coding: utf-8 -*-
+//Copyright 2018 Therp BV <https://therp.nl>
+//License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+odoo.define('web_widget_many2one_polymorphic', function(require)
+{
+    var core = require('web.core'),
+        form_relational = require('web.form_relational');
+
+    var FieldMany2OnePolymorphic = form_relational.FieldMany2One.extend({
+        init: function(field_manager, node) {
+            var self = this;
+            this._super.apply(this, arguments);
+            // reset our value when the model field changes
+            field_manager.on(
+                _.str.sprintf('field_changed:%s', this.options.model_field),
```

Comment:
Why not use string templates here?
https://github.com/lukehoban/es6features#template-strings

Reply:
We can't use ES6 features as long as we use phantomjs in our toolchain, which has a very old JS engine that would break tours and tests silently.

AFAIK in the upcoming v12, Odoo will use chrome headless to run tests, and thus we can start using XXI century JS! But for now, it's not possible, sorry :confused: